core/templates/components/question-directives/questions-list/questions-list.directive.ts

```diff
@@ -722,18 +708,7 @@ angular.module('oppia').directive('questionsList', [
                             '/components/question-directives/modal-templates/' +
                             'confirm-question-modal-exit-modal.directive.html'),
                         backdrop: true,
-                        controller: [
-                          '$scope', '$uibModalInstance',
-                          function($scope, $uibModalInstance) {
-                            $scope.cancel = function() {
-                              $uibModalInstance.dismiss('cancel');
-                            };
-
-                            $scope.close = function() {
-                              $uibModalInstance.close();
-                            };
-                          }
-                        ]
+                        controller: 'ConfirmOrCancelModalController'
                       });
                       modalInstance.result.then(function() {
```

Comment:
not sure whether this is needed, can you please check?

Reply:
We need it because this modal in being called inside another modal, and the promise is being called to handle the outer modal.

Reply:
Thanks for the explanation.